### DISCUSSION TITLE
question on pygpsclient termination of NTRIP client connection

### DISCUSSION THREAD

**Discussion by pat-2010** [semuconsulting/pygnssutils#97] [Q&A] [answered]
Troubleshooting a different problem I saw the following behavior.  Is it expected?

Configuration: Raspberry Pi 5 running pypgsclient 1.4.22. Python 3.11.42. GPS receiver not applicable, since I see this behavior without turning on my GPS receiver. EDIT: noticed that my pygpsclient version was out of date. Updated to 1.4.24 and repeated the test with no change. 

When I stop the NTRIP client using the stop button on the bottom of the NTRIP client connection, the status at the bottom says "Disconnected". Using Wireshark I can see the NTRIP message from the server and the ACK from my computer. When I disconnect, my computer continues to send ACKs and the server thinks I'm still connected. It's not until I exit pygpsclient that I get a different message from my computer that actually disconnects from the NTRIP server.

I have the wireshark stream saved in case any info there would be useful. Poking around the message stream, the last 3 messages are:

1- a 383 byte message from the server (normal transmission)
2- an ack from my computer
3- another ack from my computer, this time with the Reset bit set.

Wireshark packet dissection for the last 3 messages.  Edited for brevity. Note that 132.239.152.175 is the NTRIP server.

No.     Time               Source                Destination           Protocol Length Info
    275 10:58:00.262518025 132.239.152.175       192.168.1.80          TCP      383    ICY 200 OK  [TCP segment of a reassembled PDU]
Frame 275: 383 bytes on wire (3064 bits), 383 bytes captured (3064 bits) on interface any, id 0
 ...
    Flags: 0x018 (PSH, ACK)
        000. .... .... = Reserved: Not set
        ...0 .... .... = Accurate ECN: Not set
        .... 0... .... = Congestion Window Reduced: Not set
        .... .0.. .... = ECN-Echo: Not set
        .... ..0. .... = Urgent: Not set
        .... ...1 .... = Acknowledgment: Set
        .... .... 1... = Push: Set
        .... .... .0.. = Reset: Not set
        .... .... ..0. = Syn: Not set
        .... .... ...0 = Fin: Not set
        [TCP Flags: ·······AP···]
...

No.     Time               Source                Destination           Protocol Length Info
    276 10:58:00.262570378 192.168.1.80          132.239.152.175       TCP      68     55374 → 2104 [ACK] Seq=211 Ack=14426 Win=64128 Len=0 
Frame 276: 68 bytes on wire (544 bits), 68 bytes captured (544 bits) on interface any, id 0
...
    1000 .... = Header Length: 32 bytes (8)
    Flags: 0x010 (ACK)
        000. .... .... = Reserved: Not set
        ...0 .... .... = Accurate ECN: Not set
        .... 0... .... = Congestion Window Reduced: Not set
        .... .0.. .... = ECN-Echo: Not set
        .... ..0. .... = Urgent: Not set
        .... ...1 .... = Acknowledgment: Set
        .... .... 0... = Push: Not set
        .... .... .0.. = Reset: Not set
        .... .... ..0. = Syn: Not set
        .... .... ...0 = Fin: Not set
        [TCP Flags: ·······A····]
...

No.     Time               Source                Destination           Protocol Length Info
    277 10:58:00.373324068 192.168.1.80          132.239.152.175       TCP      68     55374 → 2104 [RST, ACK] Seq=211 Ack=14426 Win=64128 
Frame 277: 68 bytes on wire (544 bits), 68 bytes captured (544 bits) on interface any, id 0
...
    1000 .... = Header Length: 32 bytes (8)
    Flags: 0x014 (RST, ACK)
        000. .... .... = Reserved: Not set
        ...0 .... .... = Accurate ECN: Not set
        .... 0... .... = Congestion Window Reduced: Not set
        .... .0.. .... = ECN-Echo: Not set
        .... ..0. .... = Urgent: Not set
        .... ...1 .... = Acknowledgment: Set
        .... .... 0... = Push: Not set
        .... .... .1.. = Reset: Set
        .... .... ..0. = Syn: Not set
        .... .... ...0 = Fin: Not set
        [TCP Flags: ·······A·R··]
...

**Comment by semuadmin** [comment]
Install this branch of pygnssutils and try again - let me know how you get on...

```shell
 python3 -m pip install git+https://github.com/semuconsulting/pygnssutils.git@RC-1.1.5
```

<img width="800" alt="Screenshot 2024-10-28 at 08 00 31" src="https://github.com/user-attachments/assets/f1cee9b0-c948-4bfe-9d0d-7425e8d9c5a0">

**Reply by pat-2010** [reply, marked as answer]
That works. Thanks.

**Reply by semuadmin** [reply]
Thanks for confirming - fix to #98 will go into v1.1.5.

And thanks for the generous donation - very much appreciated 😃.
